app/post-contribution/route.js

```diff
@@ -17,7 +17,9 @@ export default Route.extend(AuthenticatedRouteMixin, {
   model({ date = moment().format('YYYY-MM-DD') }) {
     let user = this.get('currentUser.user')
 
-    return this.store.createRecord('contribution', { user, date })
+    let challenge = this.store.findRecord('challenge', 'current')
```

Comment:
This might be needed in the edit-contribution route too.

The current challenge might be fetched in the `beforeModel()` hook, although then we'd have to search for in in the store somehow. Might be fine as is.